dhall/src/Dhall/Syntax.hs

```diff
@@ -1097,7 +1097,10 @@ instance Pretty ImportType where
     pretty Missing = "missing"
 
 -- | How to interpret the import's contents (i.e. as Dhall code or raw text)
-data ImportMode = Code | RawText | Location
+--   OriginHeaders is identical to Code, except local imports with this mode
+--   are allowed from any source (local or remote). This type is only used when
+--   loading Origin headers, it can't be set by the user.
+data ImportMode = OriginHeaders | Code | RawText | Location
```

Comment:
I don't think we need a new `ImportMode` if we implement this in the same way as the standard specifies.  I'll comment elsewhere on how I think we can make this work without the `OriginHeaders` mode